tests/ValidatorDelegates/CSPBadFlagsTest.php

```diff
@@ -0,0 +1,182 @@
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "default-src 'unsafe-inline' 'unsafe-eval'; "
+                . "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
```

Comment:
This might be a good place to use [named constructors](http://verraes.net/2014/06/named-constructors-in-php/) instead of reimplementing the method under test in the test. :)